src/sentry/integrations/gitlab/client.py

```diff
@@ -64,59 +76,40 @@ def get_group(self, group):
         path = GitLabApiClientPath.group.format(group=group)
         return self.get(path)
 
-    def request(self, method, path, data=None, params=None):
-        headers = {"Authorization": f"Bearer {self.token}"}
-        return self._request(
-            method,
-            GitLabApiClientPath.build_api_url(self.base_url, path),
-            headers=headers,
-            data=data,
-            params=params,
-        )
-
 
-class GitLabApiClient(ApiClient):
+class GitLabProxyApiClient(IntegrationProxyClient):
     integration_name = "gitlab"
 
     def __init__(self, installation):
         self.installation = installation
         verify_ssl = self.metadata["verify_ssl"]
         self.is_refreshing_token = False
+        self.new_identity = None
+        self.base_url = self.metadata["base_url"]
         super().__init__(verify_ssl)
 
     @property
     def identity(self):
-        return self.installation.default_identity
+        if self.new_identity:
+            return self.new_identity
```

Comment:
The identity isn't really 'new' it has just had its tokens replaced. Perhaps `updated_identity` or `refreshed_identity` could work?